### ISSUE TITLE
Take Fee on withdraw

### ISSUE THREAD

**Issue by jonn4y** [3s3s/opentrade#117]
Hey @3s3s I have tried following the code back as far as i can to see where the coin.info.hold is taken and given back but im struggling to find it. when i make a withdraw i see 3 transactions

<img width="1200" alt="screen shot 2018-09-05 at 14 59 07" src="https://user-images.githubusercontent.com/726705/45098140-4636d780-b11c-11e8-9c79-85ac9aff38cb.png">

the last one as you can see is the coin hold fee being given back but instead of going back into the users account i am wanting it to go into the root account, any idea where that would be?

**Comment by knkrth** [comment]
@3s3s Yes provide a option on admin page in the coins section `Fee Address:` so that we can provide our own coin address. All the fee can be moved to the provided address.

**Comment by coinnetworkpw** [comment]
@jonn4y 

Its from this line - https://github.com/3s3s/opentrade/blob/master/server/modules/users/wallet.js#L847
to the end

**Comment by knkrth** [comment]
@coinnetworkpw Did you make changes in your exchange? If so can create a pull request with those changes or can you share your source code with us email me gsrwallet@gmail.com

**Comment by coinnetworkpw** [comment]
No I didnt make a changes, I dont need that feature on my exchange

**Comment by jonn4y** [comment]
I have it now giving the fee to the specified account from within the constants file (in my case userid1 gets the fees) and it is doing it on the wallets just need to get it doing it via the database now to reflect in the users accounts then I will share the code

**Comment by knkrth** [comment]
@jonn4y Ok.

**Comment by TheRealHotSwap** [comment]
@jonn4y mind sharing your changes?

**Comment by CryptorClub** [comment]
@3s3s in latest commit fee is deducted from clients balance, but not written in DB

**Comment by jonn4y** [comment]
Yes I just need to write the fee to my root accounts dB entry so my userID1 takes all the fees I will get round to this maybe today

**Comment by hashbook** [comment]
@jonn4y any progress on this modification? if you don't have time it would be great if you can publish what you did so far so i can take a look. tnx.

**Comment by jonn4y** [comment]
Hey,
I added my code to my live exchange and it caused issued when people were sending deposits it was also crediting the root account which messed up the balances so i removed them.
but this is sort of where i was looking

in constants.js i added:
`exports.ExchangeFeeAccountID = 1;` 
as thats the account i wanted all fees to goto

then
https://github.com/3s3s/opentrade/blob/master/server/modules/users/wallet.js#L401

i changed to:
`MoveBalance(userID, g_constants.ExchangeFeeAccountID, coin, ret.data, err => {`

that would add the fee to my account userid 1 as specified in the constants.js
but that wouldn't save to the DB

the DB stuff is from 
https://github.com/3s3s/opentrade/blob/master/server/modules/users/wallet.js#L847
as mentioned by @coinnetworkpw above

my stuff was deleted when i panicked that all of a sudden the root account was filling up with 1000's of coins so i revered

**Comment by CryptorClub** [comment]
@jonn4y, If it's works correctly, please add a pull request

**Comment by jonn4y** [comment]
Didn’t you read my message lol the extra code caused all deposits go into my root acc
The dirt but works moving the fee to root account but then it automatically moves everything back to the main account

**Comment by jonn4y** [comment]
Hey @hashbook did you manage to take a look at it?

**Comment by Firoz81** [comment]
@jonn4y 

I haven't studied this code deeply but i have tested this code on server.

Adding of following in constants.js
exports.ExchangeFeeAccountID = 1;

and changing of following
MoveBalance(userID, g_constants.ExchangeFeeAccountID, coin, ret.data, err => {

above changes will obviously send all balances to UserID1=admin.

You have to define in code that upon withdraw success blocked amount for fee moves to UserID1=admin or contants.js default ExchangeBalanceAccountID=0. 

Try to fix code to move fee amount to default exchange account first, that will take fee amount from user and move to account 0, fee will not go to UserID1=admin but fee will be taken from users withdraws and will remain in wallet main account.

When I'll have time I will try again on test server changing lines in server/modules/users/wallet.js for fees to move to default exchange account upon success of withdraw.

Meanwhile if you find solution please post here that will save my time and efforts.
